src/Symfony/Component/Form/Extension/Validator/ValidatorTypeGuesser.php

```diff
@@ -122,7 +122,12 @@ public function guessTypeForConstraint(Constraint $constraint)
 
             case 'Symfony\Component\Validator\Constraints\File':
             case 'Symfony\Component\Validator\Constraints\Image':
-                return new TypeGuess('Symfony\Component\Form\Extension\Core\Type\FileType', [], Guess::HIGH_CONFIDENCE);
+                $options = [];
+                if (null !== $constraint->mimeTypes && !empty($constraint->mimeTypes)) {
```

Comment:
Please do not use `empty()`.  Let's use something like this instead:

```php
$options = [];
$mimeTypes = (array) $constraint->mimeTypes;

if ($mimeTypes) {
    $options = [
        'attr' => [
            'accept' => implode(',', $mimeTypes),
        ],
    ];
}
```

Reply:
Hello,

If I pass empty string in mimeTypes, this condition not work because mimeTypes has one key with empty string value.

```php
/**
 * @var File|null image
 *
 * @Assert\File(mimeTypes="")
 */
```

Is interesting to test if first value is not empty ?

```
$options = [];
$mimeTypes = (array) $constraint->mimeTypes;
if ($mimeTypes && !empty($mimeTypes[key($mimeTypes)])) {
    $options = ['attr' => ['accept' => implode(',', $mimeTypes)]];
}
```

Reply:
I think we shouldn't care about that case. Using an empty string looks like a misconfigured constraint to me instead.

Reply:
But, changing my suggestion to something like that should cover that too but still avoids using `empty`:

```php
$options = [];

if ($constraint->mimeTypes) {
    $options = [
        'attr' => [
            'accept' => implode(',', (array) $constraint->mimeTypes),
        ],
    ];
}
```

Reply:
Thank you for the advice regarding the "empty" ;)
I updated my code to use your suggestion.